hrc_speech_prediction/evaluate_tree_model.py

```diff
@@ -0,0 +1,373 @@
+            test_idx = [j for j in range(i, next_i)]
+            train_idx = [j for j in range(0, n_samples) if j not in test_idx]
+            results.append(self._evaluate_on(train_idx, test_idx, data_type))
+        self._print_global_results(results)
+
+    def new_participant(self, data_type="context"):
+        print("Testing on pilot participant...")
+        results = {}
+        trials = ["A", "D", "T"]
+        for t in trials:
+            train_idx = [
+                i for p in TRAIN_PARTICIPANTS for i in self.data.data[p].ids
+            ]
+            test_idx = [
+                i
+                for part in self.test_participants
+                for trial in self.data.data[part] for i in trial.ids
+                if t == trial.instruction
+            ]
+            results[t] = self._evaluate_on(train_idx, test_idx, data_type)
+        self._print_result_table(results, 'Instruction')
+
+    def test_all(self):
+        for data_type in ["context", "speech", "both"]:
+            print("---------------testing on {}---------------"
+                  .format(data_type))
+            self.test_on_one_participant(data_type)
+            self.test_on_one_trial(data_type)
+            self.cross_validation(data_type)
+            self.new_participant(data_type)
+
+    def test_incremental_learning(self, data_type):
+        classes = np.unique((self.data.labels))
+        score = []
+        for i in range(0, self.X_context.shape[0] - 1):
+            # Train set
+            train_X = self.get_Xs([i])
+            train_Y = self.get_labels([i])
+            # Test set
+            test_X = self.get_Xs([i + 1])
+            test_Y = self.get_labels([i + 1])
+            model = self.model(self.context_actions,
+                               features=data_type,
+                               randomize_context=.25).partial_fit(train_X[0],
+                                                                  train_X[1],
+                                                                  train_Y,
+                                                                  classes)
+            prediction = model.predict(*test_X)
+            score.append(metrics.accuracy_score(
+                test_Y, prediction, normalize=True, sample_weight=None))
+
+        print(score)
+        xaxis = np.array([i for i in range(0, self.X_context.shape[0] - 1)])
+        score = np.array(score)
+
+        xnew = np.linspace(xaxis.min(), xaxis.max(), 1000)
+        score_smooth = spline(xaxis, score,xnew)
+
+        plt.plot(xnew, score_smooth)
+        plt.show()
+
+
+    def _evaluate_on(self, train_idx, test_idx, data_type, speech_eps):
+        # self.check_indices(train_idx, test_idx)
+
+        flat_train_idx = [i for p in train_idx for t in p for i in t]
+        flat_test_idx = [i for p in test_idx for t in p for i in t]
+        test_utters = [list(self.data.utterances)[i] for i in flat_test_idx]
+        self.Us += [i for i in test_utters]
```

Comment:
I guess you store them in order to be able to retrieve them for the plot. It would make more sense here to just return a more complete object as the result that includes the ids used. Typically at the end of running a test you return a dictionary like `{'scores': ..., 'test_ids': ..., 'test_utterances': ...}`. It could be a tuple or a dedicated class (that does not matter) but you generally want to retain more information from a run. That way you can also analyse results in `ipython` without having to re-run the code every time.